File: sdk/dotnet/lmsservice_db/src/api/course.content/CourseContentModel.cs

using System.ComponentModel.DataAnnotations;

namespace lmsservice_db.src.api.course.content.course.contentmodel;
public class CourseContentCreateModel
{
        [Required]
    [StringLength(64, MinimumLength = 0)]
    public string Title { get; set; } = string.Empty;

    [Required]
    [StringLength(64, MinimumLength = 0)]
    public string Description { get; set; } = string.Empty;

    [Required]
    [StringLength(64, MinimumLength = 0)]
    public string ImageUrl { get; set; } = string.Empty;

    [Required]
    [StringLength(64, MinimumLength = 0)]
    public int DurationInMins { get; set; } = 0;

    [Required]
    [StringLength(64, MinimumLength = 0)]
    public string ContentType { get; set; } = string.Empty;

    [Required]
    [StringLength(64, MinimumLength = 0)]
    public string ResourceLink { get; set; } = string.Empty;

    [Required]
    [StringLength(64, MinimumLength = 0)]
    public string ActionTemplateId { get; set; } = string.Empty;

    [Required]
    [StringLength(64, MinimumLength = 0)]
    public int Sequence { get; set; } = 0;


}

public class CourseContentUpdateModel
{
        [Required]
    [StringLength(64, MinimumLength = 0)]
    public string Title { get; set; } = string.Empty;

    [Required]
    [StringLength(64, MinimumLength = 0)]
    public string Description { get; set; } = string.Empty;

    [Required]
    [StringLength(64, MinimumLength = 0)]
    public string ImageUrl { get; set; } = string.Empty;

    [Required]
    [StringLength(64, MinimumLength = 0)]
    public int DurationInMins { get; set; } = 0;

    [Required]
    [StringLength(64, MinimumLength = 0)]
    public string ContentType { get; set; } = string.Empty;

    [Required]
    [StringLength(64, MinimumLength = 0)]
    public string ResourceLink { get; set; } = string.Empty;

    [Required]
    [StringLength(64, MinimumLength = 0)]
    public string ActionTemplateId { get; set; } = string.Empty;

    [Required]
    [StringLength(64, MinimumLength = 0)]
    public int Sequence { get; set; } = 0;


}

public class CourseContentSearchFilters
{
        [StringLength(64, MinimumLength = 0)]
    public string Title { get; set; } = string.Empty;

    [StringLength(64, MinimumLength = 0)]
    public string Description { get; set; } = string.Empty;

    [StringLength(64, MinimumLength = 0)]
    public string ImageUrl { get; set; } = string.Empty;

    [StringLength(64, MinimumLength = 0)]
    public int DurationInMins { get; set; } = 0;

    [StringLength(64, MinimumLength = 0)]
    public string ContentType { get; set; } = string.Empty;

    [StringLength(64, MinimumLength = 0)]
    public string ResourceLink { get; set; } = string.Empty;

    [StringLength(64, MinimumLength = 0)]
    public string ActionTemplateId { get; set; } = string.Empty;

    [StringLength(64, MinimumLength = 0)]
    public int Sequence { get; set; } = 0;


}

public class ApiResponse
{
    [Required]
    public Dictionary<string, object>? Data { get; set; }

    [StringLength(500, MinimumLength = 2)]
    public string Message { get; set; } = string.Empty;

    public bool? Success { get; set; }
}

File: sdk/node/src/api/course.content/course.content.model.ts

import { IsString, IsNumber, IsOptional, IsBoolean, IsUUID } from 'class-validator';

export class CourseContentCreateModel {
    @IsOptional()
    @IsUUID()
    id?: string;  
    @IsString()
    Title: string;
    @IsString()
    Description: string;
    @IsString()
    ImageUrl: string;
    @IsNumber()
    DurationInMins: number;
    @IsString()
    ContentType: string;
    @IsString()
    ResourceLink: string;
    @IsString()
    ActionTemplateId: string;
    @IsNumber()
    Sequence: number;

}

export class CourseContentUpdateModel {
    @IsOptional()
    @IsString()
    Title?: string;
    @IsOptional()
    @IsString()
    Description?: string;
    @IsOptional()
    @IsString()
    ImageUrl?: string;
    @IsOptional()
    @IsNumber()
    DurationInMins?: number;
    @IsOptional()
    @IsString()
    ContentType?: string;
    @IsOptional()
    @IsString()
    ResourceLink?: string;
    @IsOptional()
    @IsString()
    ActionTemplateId?: string;
    @IsOptional()
    @IsNumber()
    Sequence?: number;

}

export class CourseContentSearchParams {
    @IsString()
    name: string;
}

export class ApiResponse<T> {
    Data: T;

    @IsOptional()
    @IsString()
    Message?: string;

    @IsOptional()
    @IsBoolean()
    Success?: boolean;
}